porch/apiserver/pkg/registry/porch/background.go

```diff
@@ -151,3 +163,38 @@ func (b *background) cacheRepository(ctx context.Context, repo *configapi.Reposi
 	}
 	return nil
 }
+
+type backoffTimer struct {
+	min, max, curr time.Duration
+	timer          *time.Timer
+}
+
+func newBackoffTimer(min, max time.Duration) *backoffTimer {
```

Comment:
There is a backoff somewhere in kubernetes apimachinery, but IIRC it's a bit convoluted

Reply:
oh, I think I found something I can use ... stay tuned.

Reply:
decided to leave this as-is for now. the apimachiner backoff could be used but would require additional code to support reset and all that is not worth it.